go/createbucket.go

```diff
@@ -0,0 +1,103 @@
+package bucketclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type createBucketOptionSet struct {
+	sessionId      int
+	makeIdempotent bool
+}
+
+type CreateBucketOption func(*createBucketOptionSet)
+
+func CreateBucketSessionIdOption(sessionId int) CreateBucketOption {
+	return func(options *createBucketOptionSet) {
+		options.sessionId = sessionId
+	}
+}
+
+func CreateBucketMakeIdempotent(options *createBucketOptionSet) {
+	options.makeIdempotent = true
+}
+
+// CreateBucket creates a bucket in metadata.
+// bucketAttributes is a JSON blob of bucket attributes
+// opts is a set of options:
+//
+//	CreateBucketSessionIdOption forces the session ID where the bucket to be
+//	    created will land
+//
+//	CreateBucketMakeIdempotent makes the request return a success if a bucket
+//	    with the same UID already exists (otherwise returns 409 Conflict, as
```

Comment:
Just wondering: how to we set the uid field ?
Is it something we already had or we added for the migration bucketd ?